instrumentation/spring/spring-webflux-5.0/javaagent/src/test/java/SpringWebfluxTest.java

```diff
@@ -0,0 +1,841 @@
+                            equalTo(NET_TRANSPORT, IP_TCP),
+                            equalTo(NET_SOCK_PEER_ADDR, "127.0.0.1"),
+                            satisfies(NET_SOCK_PEER_PORT, val -> val.isInstanceOf(Long.class)),
+                            equalTo(NET_SOCK_HOST_ADDR, "127.0.0.1"),
+                            equalTo(NET_HOST_NAME, "localhost"),
+                            satisfies(NET_HOST_PORT, val -> val.isInstanceOf(Long.class)),
+                            equalTo(HTTP_TARGET, parameter.urlPath),
+                            equalTo(HTTP_METHOD, "GET"),
+                            equalTo(HTTP_STATUS_CODE, 200),
+                            equalTo(HTTP_SCHEME, "http"),
+                            equalTo(HTTP_FLAVOR, "1.1"),
+                            satisfies(HTTP_USER_AGENT, val -> val.isInstanceOf(String.class)),
+                            equalTo(HTTP_ROUTE, parameter.urlPathWithVariables),
+                            satisfies(
+                                HTTP_REQUEST_CONTENT_LENGTH,
+                                val ->
+                                    val.satisfiesAnyOf(
+                                        v -> assertThat(v).isInstanceOf(Long.class),
+                                        v -> assertThat(v).isNull())),
+                            satisfies(
+                                HTTP_RESPONSE_CONTENT_LENGTH,
+                                val ->
+                                    val.satisfiesAnyOf(
+                                        v -> assertThat(v).isInstanceOf(Long.class),
+                                        v -> assertThat(v).isNull()))),
+                span -> {
+                  if (parameter.annotatedMethod == null) {
+                    // Functional API
+                    assertThat(trace.getSpan(1).getName())
+                        .contains(SPRING_APP_CLASS_ANON_NESTED_CLASS_PREFIX, ".handle");
+                  } else {
+                    // Annotation API
+                    span.hasName(
+                        TestController.class.getSimpleName() + "." + parameter.annotatedMethod);
+                  }
+                  span.hasKind(SpanKind.INTERNAL)
+                      .hasParent(trace.getSpan(0))
+                      .hasAttributesSatisfyingExactly(
+                          satisfies(
+                              stringKey("spring-webflux.handler.type"),
+                              parameter.annotatedMethod == null
+                                  ? val -> val.contains(INNER_HANDLER_FUNCTION_CLASS_TAG_PREFIX)
+                                  : val -> val.isEqualTo(TestController.class.getName())));
+                });
+
+    List<Consumer<TraceAssert>> traceAssertions = new ArrayList<>();
+    IntStream.rangeClosed(0, requestsCount - 1)
+        .forEachOrdered(n -> traceAssertions.add(traceAssertion));
+
+    testing.waitAndAssertTraces(traceAssertions);
```

Comment:
Nit: could be simplified
```suggestion
    testing.waitAndAssertTraces(Collections.nCopies(requestsCount, traceAssertion));
```

Reply:
thank you :)